public/app.js

```diff
@@ -216,5 +217,36 @@
         });
 
         $httpProvider.interceptors.push('jwtInterceptor');
-    }
+
+        $translateProvider
+            .useSanitizeValueStrategy('sanitizeParameters')
+            .useStaticFilesLoader({
+                prefix: 'static/languages-<%= VERSION %>/',
+                suffix: '.json'
+            })
+            .determinePreferredLanguage(function () {
+                let lang = 'en'; // default language
+
+                if (navigator.language) {
+                    lang = navigator.language;
+                }
+
+                // if it is set via query, use that
+                const queries = document.location.search.substr(1).split('&');
```

Comment:
I've been using `$location` elsewhere to get query parameters - e.g. https://github.com/IBM/taxinomitis/blob/8f98a62a0b394871b7dd6bcbd66b86e8b403d6b3/public/components/login/login.controller.js#L69 

Is there any difference/benefit to either approach?

Reply:
Oh, yeah I forgot about `$location`, you don't get that in React, you only get the standard js API!

Reply:
welcome back to Angular loveliness ;-)

Reply:
Actually, saying that, it's not working for me. I don't think you can use `$location` inside `angular.config`

Reply:
https://stackoverflow.com/questions/18137662/why-cant-i-get-a-location-injected-into-in-my-config